### ISSUE TITLE
"screen is not defined" when used with PROXY_TO_PTHREAD

### ISSUE THREAD

**Issue by lincoln-b** [emscripten-ports/SDL2#110]
When compiled with `-s PROXY_TO_PTHREAD=1`, calls to `SDL_Init()` fail with the error "screen is not defined".

This is important to me because proxying all socket commands through a local websocket server [as described here](https://emscripten.org/docs/porting/networking.html#full-posix-sockets-over-websocket-proxy-server) is the only way I can get any networking to work in my project.

Here's my source file, poc.c:

```
#include "SDL.h"

int main() {
  SDL_Init(SDL_INIT_VIDEO);
  SDL_Quit();
}
```

And here's my commands to compile and run:

```
emcc poc.c -s USE_SDL=2 -s USE_PTHREADS=1 -s PROXY_POSIX_SOCKETS=1 -s PROXY_TO_PTHREAD=1 -o poc.html
python -m SimpleHTTPServer 5000
```

Open localhost:5000/poc.html in a browser, open the developer console, and you see this error:

```
poc.worker.js:175 Uncaught ReferenceError: screen is not defined
    at 37846 (http://localhost:2222/poc.js:1908:21)
    at _emscripten_asm_const_iii (http://localhost:2222/poc.js:1927:27)
    at wasm-function[968]:0x6a84c
    at wasm-function[852]:0x603e7
    at wasm-function[260]:0x3472
    at wasm-function[261]:0x35b0
    at wasm-function[258]:0x339a
    at wasm-function[259]:0x33a9
    at Module._main (http://localhost:2222/poc.js:9838:32)
    at ___call_main (http://localhost:2222/poc.js:2312:24)
```

From jakogut's comments [on this site](https://gitmemory.com/issue/emscripten-core/emscripten/6009/468017397), it appears that the problem is simply that the screen object is not proxied to the new main thread. Relevant lines from `video/emscripten/SDL_emscriptenvideo.c`:

```
mode.w = EM_ASM_INT_V({
    return screen.width;
}); 

mode.h = EM_ASM_INT_V({
    return screen.height;
}); 
```

Indeed, the documentation on PROXY_TO_PTHREAD only says that the canvas object is proxied to the new main thread. How can we get `screen` proxied as well? I'm happy to contribute code, just need to be pointed in the right direction!

**Comment by Daft-Freak** [comment]
Hmm, there was some discussion on this here #77 and some work on new APIs here https://github.com/emscripten-core/emscripten/pull/9336/.

The quickest solution is probably to switch to `MAIN_THREAD_EM_ASM_INT`. (Though a get_screen_size API would still be nice...)

**Comment by lincoln-b** [comment]
Oh nice, glad to see it's being worked on. I'll just a workaround for now!

**Comment by danielknobe** [comment]
This commits solves the problem but hidden error become visible.
The GL Context (GLctx) is undefined, when I try to run this short example:
```

**Comment by Daft-Freak** [comment]
Yeah, there are still other issues, that one can be fixed with `-s OFFSCREEN_FRAMEBUFFER=1`, but then you hit https://github.com/emscripten-core/emscripten/issues/7684.

**Comment by danielknobe** [comment]
Yes this fixes that problem. Ty.
